prov/gni/src/gnix_ep.c

```diff
@@ -576,6 +593,12 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 		}
 	}
 
-	if (nic != NULL) free(nic);
+	pthread_mutex_lock(&gnix_nic_list_lock);
+	if (incremented_ptag_nic_cnt)
```

Comment:
I'm not sure I understand the purpose of this 'incremented_ptag_nic_cnt' variable. Could you explain?